src/qibo/backends/_clifford_operations.py

```diff
@@ -164,6 +164,17 @@ def RY(symplectic_matrix, q, nqubits, theta):
         return RY_3pi_2(symplectic_matrix, q, nqubits)
 
 
+def GPI2(symplectic_matrix, q, nqubits, phi):
+    if phi % (2 * np.pi) == 0:
+        return RX(symplectic_matrix, q, nqubits, np.pi / 2)
+    elif (phi / np.pi - 1) % 2 == 0:
+        return RX(symplectic_matrix, q, nqubits, -np.pi / 2)
+    elif (phi / (np.pi / 2) - 1) % 4 == 0:
+        return RY(symplectic_matrix, q, nqubits, np.pi / 2)
+    else:  # theta == 3*pi/2 + 2*n*pi
+        return RY(symplectic_matrix, q, nqubits, -np.pi / 2)
```

Comment:
```suggestion
def GPI2(symplectic_matrix, q, nqubits, phi):
    if phi % (2 * np.pi) == 0:
        return RX(symplectic_matrix, q, nqubits, np.pi / 2)
    
    if (phi / np.pi - 1) % 2 == 0:
        return RX(symplectic_matrix, q, nqubits, -np.pi / 2)
    
    if (phi / (np.pi / 2) - 1) % 4 == 0:
        return RY(symplectic_matrix, q, nqubits, np.pi / 2)
    
    # theta == 3*pi/2 + 2*n*pi
    return RY(symplectic_matrix, q, nqubits, -np.pi / 2)
```

What happens when the choice of angle leads to a non-Clifford GPI2? It should not work here.

Reply:
I think the point is that these gates are only used in the `CliffordBackend.apply_gate_clifford` method, so it's not necessary for them to work when the gate is not Clifford. Take a look, for example, at the implementation of other gates like `RX` or `RZ` in the `_clifford_operations.py` file.

Reply:
I see, the check if it is Clifford or not already happened.